example/fft_ex01.cpp

```diff
@@ -25,12 +25,12 @@ int main()
     std::vector< std::complex<double> > A{1.0,2.0,3.0,4.0},B(A.size());
     
     // default fft engine, forward transform, out-of-place
-    boost::math::fft::dft_forward(A.cbegin(),A.cend(),B.begin());
+    fft::transform<fft::bsl_dft<std::complex<double>>>::forward(A.cbegin(),A.cend(),B.begin());
```

Comment:
A comment on the new API: I wanted the `transform` container to be template on a type and not on a template type. That is, you pass a template argument one of the DFT engines already specialized, for instance
```
transform<bsl_dft<std::complex<double>>>::forward(A.cbegin(),A.cend(),B.begin());
```
Internally, however, the `transform` will statically select the DFT engine version according to the arguments:
https://github.com/BoostGSoC21/math/blob/354f44f053b7471b662c97b5eaf374eb137c099b/include/boost/math/fft/dft_api.hpp#L177-L182

that is possible for every back-end because the code `other` is implemented in this line of the front-end:
https://github.com/BoostGSoC21/math/blob/354f44f053b7471b662c97b5eaf374eb137c099b/include/boost/math/fft/dft_api.hpp#L48-L49

Then, because we don't actually use the `std::complex<double>`, you could transform using the default arguments of `bsl_dft` in this way:
```
transform< bsl_dft<>>::forward(A.cbegin(),A.cend(),B.begin());
```
or using this alias:
https://github.com/BoostGSoC21/math/blob/354f44f053b7471b662c97b5eaf374eb137c099b/include/boost/math/fft/bsl_backend.hpp#L218

the transform line becomes
```
bsl_transform::forward(A.cbegin(),A.cend(),B.begin());
```

Reply:
Hi @Lagrang3 , this is the exact kind of aliases which I had in mind. The template deduction done in this part is very nice!